lib/srv/db/common/engines.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/trace"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// engines is a global database engines registry.
+	engines map[string]EngineFn
+	// enginesMu protects access to the global engines registry map.
+	enginesMu sync.RWMutex
+)
+
+// EngineFn defines a database engine constructor function.
+type EngineFn func(EngineConfig) (Engine, error)
```

Comment:
The error result from all common.RegisterEngine calls seems to be always hardcoded to nil thus right now seems to be redundant or MSSQL engine somehow is going to leverage this error param ?

Reply:
Yeah I thought it could be useful but none of existing implementers actually use it. I've removed it, we can always add later if needed.